test/lib/failOnError.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var assert = require('assert');
+var AwsHelper = require('./../../lib/index');
+var simple = require('simple-mock');
+
+describe('AwsHelper.failOnError', function (done) {
+  afterEach(function () {
+    simple.restore();
+  });
+
+  it('should do nothing if there is no error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var spy = simple.spy(fakeContext, 'fail');
```

Comment:
I have used the spy to ensure that the function is not called. 
I could have done that like:
```js 
    var fakeContext = {
      fail: simple.spy(function () {})
    };
```
But I thought it was more verbose the other way.